CRM/Upgrade/Incremental/php/FiveTwentyEight.php

```diff
@@ -26,9 +26,15 @@ class CRM_Upgrade_Incremental_php_FiveTwentyEight extends CRM_Upgrade_Incrementa
    */
   public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NULL) {
     // Example: Generate a pre-upgrade message.
-    // if ($rev == '5.12.34') {
-    //   $preUpgradeMessage .= '<p>' . ts('A new permission, "%1", has been added. This permission is now used to control access to the Manage Tags screen.', array(1 => ts('manage tags'))) . '</p>';
-    // }
+    if (!function_exists('civi_wp')) {
+      //exit
+    }
+    elseif ($rev == '5.28.alpha1') {
+      $preUpgradeMessage .= '<p>' . ts('Starting with version 5.29.0, CiviCRM on WP will
+        automate the determination of the civicrm files directory. If you have a non-standard
+        directory structure you will need to either override the settings in civicrm.settings.php
+         or by specifying the locations in System Settings Directories and System Settings Resource URLs.') . '</p>';
```

Comment:
```suggestion
         or by specifying the locations in System Settings Directories and System Settings Resource URLs.') . Starting with version 4.??.??, wp-content/uploads/civicrm/ isthe standard WordPress CiviCRM Files directory. '</p>';
```

Reply:
While `r-run`ning this PR, I tried on `4.6` (branch) and `4.7.31` (tag). For `4.6`, it gave the old layout. For `4.7.31`, it gave the new layout. (I didn't check to see if the change was 4.7.0 or mid-4.7.x.) I would just say it changed in 4.7.

FWIW, this is the command that gave me the old layout: 

```
civibuild create wp46 --civi-ver 4.6 --cms-ver 4.9.6
```

(Aside: This required a little bit of hacking at my environment to get php56 for Civi 4.6 -- and to get php71 for running the upgrade.)

Reply:
The change was in 2015 4.7.0 - https://github.com/civicrm/civicrm-wordpress/commit/36ca543653d54f869d3698716fe5135fcb6e314e